File: light-weights-be/src/main/java/com/maksymilianst/lightweights/plan/validator/TrainingValidator.java
package com.maksymilianst.lightweights.plan.validator;


import com.maksymilianst.lightweights.plan.dto.TrainingDto;
import com.maksymilianst.lightweights.plan.model.TrainingBlock;


public interface TrainingValidator {

    void validate(TrainingDto trainingDto, TrainingBlock block);
}

File: light-weights-fe/src/app/plans/validators/training.validator.ts
import {AbstractControl, ValidationErrors, ValidatorFn} from '@angular/forms';
import {TrainingBlock} from '../models/training-plan.model';

export class TrainingValidator {

  public static trainingDateInBlockRange(block: TrainingBlock): ValidatorFn {
    return (control: AbstractControl): ValidationErrors | null => {
      const trainingDate = control.value;

      return block.start <= trainingDate && trainingDate <= block.end
        ? null
        : {endDateAfterStartDate: true};
    }
  }

}
